src/main/scala/io/qbeast/table/IndexedTable.scala

```diff
@@ -524,45 +569,92 @@ private[table] class IndexedTableImpl(
     }
   }
 
-  override def optimize(revisionID: RevisionID, options: Map[String, String]): Unit =
-    optimize(revisionID, 1.0, options)
+  override def optimizeUnindexedFiles(
+      unindexedFiles: Seq[String],
+      options: Map[String, String]): Unit = {
+    if (unindexedFiles.isEmpty) return // Nothing to optimize
+    // 1. Load the files from the Staging ID (Unindexed)
+    val files =
```

Comment:
Convert `unindexedFiles` to a Set first to reduce time complexity.

Reply:
done in 8cab2b8